src/impl.jl

```diff
@@ -221,7 +224,19 @@ function auto_hash_equals_impl(__source__, struct_decl, fields, cache::Bool, has
         push!(type_body, :(_cached_hash::UInt))
 
         # Add the internal constructor
-        hash_init = (typearg && !isnothing(where_list)) ? :($hashfn($full_type_name)) : :($hashfn($(QuoteNode(type_name))))
+        hash_init = if isnothing(typeseed)
+            if typearg
+                :($hashfn($full_type_name))
+            else
+                :($hashfn($(QuoteNode(type_name))))
+            end
+        else
+            if typearg
+                :(UInt($typeseed($full_type_name)))
+            else
+                :(UInt($typeseed))
```

Comment:
it feels a little weird to me that `typeseed` is sometimes a scalar and sometimes a function. 

Did you consider always making it a function and then calling it with `type_name` vs `full_type_name`? Though the ergonomics of that might be annoying. 

Or maybe making it a scalar or callable, but then changing the callable argument depending on `typearg`? Though then I suppose the equality guarantees of typearg are at risk 🤔 

Perhaps it's fine as is, though maybe in the docs just make it really clear that you expect a number in one case and a function in the other. I think right now that's a little subtle and could catch people up

Reply:
I've clarified the docs.